movingpandas/trajectory_smoother.py

```diff
@@ -0,0 +1,194 @@
+    from stonesoup.types.array import CovarianceMatrix, StateVector
+    from stonesoup.types.prediction import GaussianStatePrediction
+    from stonesoup.initiator.simple import SimpleMeasurementInitiator
+    from stonesoup.types.hypothesis import SingleHypothesis
+    from stonesoup.smoother.kalman import KalmanSmoother
+    from stonesoup.reader.base import DetectionReader
+    from stonesoup.base import Property
+    from stonesoup.buffered_generator import BufferedGenerator
+except ImportError as error:
+    raise ImportError(
+        "Missing optional dependencies. To use the smoother classes please install movingpandas along with the "
+        "'smoothing' optional dependencies (e.g. pip install 'movingpandas[smoothing]')."
+    ) from error
+
+from .trajectory import Trajectory
+from .trajectory_collection import TrajectoryCollection
+
+
+class TrajectorySmoother(ABC):
+    """
+    TrajectorySmoother base class
+
+    Base class for trajectory smoothers. This class is abstract and thus cannot be instantiated.
+    """
+    def __init__(self, traj):
+        """
+        Create TrajectorySmoother
+
+        Parameters
+        ----------
+        traj : Trajectory or TrajectoryCollection
+        """
+        self.traj = traj
+
+    def smooth(self, **kwargs):
+        """
+        Smooth the input Trajectory/TrajectoryCollection
+
+        Parameters
+        ----------
+        kwargs : any type
+            Keyword arguments, differs by smoother
+
+        Returns
+        -------
+        Trajectory/TrajectoryCollection
+            Smoothed Trajectory or TrajectoryCollection
+
+        """
+        if isinstance(self.traj, Trajectory):
+            return self._smooth_traj(self.traj, **kwargs)
+        elif isinstance(self.traj, TrajectoryCollection):
+            return self._smooth_traj_collection(**kwargs)
+        else:
+            raise TypeError
+
+    def _smooth_traj_collection(self, **kwargs):
+        smoothed = []
+        for traj in self.traj:
+            smoothed.append(self._smooth_traj(traj, **kwargs))
+        result = copy(self.traj)
+        result.trajectories = smoothed
+        return result
+
+    @abstractmethod
+    def _smooth_traj(self, traj, **kwargs):
+        raise NotImplementedError
+
+
+class KalmanSmootherCV(TrajectorySmoother):
+    """
+    Smooths using a Kalman Filter with a Constant Velocity model.
+
+    The Constant Velocity model assumes that the speed between consecutive locations is nearly
+    constant. For trajectories where ``traj.is_latlon = True`` the smoother converts
+    to EPSG:3395 (World Mercator) internally to perform filtering and smoothing.
+
+    .. note::
+        This class makes use of `Stone Soup <https://stonesoup.readthedocs.io/en/v0.1b7/>`_, which is an
+        optional dependency and not installed by default. To use this class, you need to install Stone Soup
+        directly (see `here <https://stonesoup.readthedocs.io/en/v0.1b7/#installation>`_).
```

Comment:
Let's avoid linking a specific version and use latest instead
https://stonesoup.readthedocs.io/en/latest/#installation

Reply:
Good spot! 👍